### PR TITLE
Allow google.project.id for passing project to resource_project_service

### PR DESCRIPTION
**Pull request by ScottSuarez**
Added ability to pass google.project.id to google.project_service.project

Fixes [7203](https://github.com/hashicorp/terraform-provider-google/issues/7203)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
project_service: Added ability to pass google.project.id to project property
```

### CODE REVIEW

File: third_party/terraform/resources/resource_google_project_service.go [hunk 1]
```diff
@@ -82,10 +82,11 @@ func resourceGoogleProjectService() *schema.Resource {
...
 			"project": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
-				ForceNew: true,
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareResourceNames,
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_google_project_service.go:89]: `compareSelfLinkOrResourceName` is more typical- any specific reason you chose this one over that?

File: third_party/terraform/resources/resource_google_project_service.go [hunk 2]
```diff
@@ -102,6 +103,13 @@ func resourceGoogleProjectService() *schema.Resource {
 	}
 }
 
+// Since we allow the project to now come from config in the form of
+// /projects/${myproject} we need to correct for this when reading the value.
+func adjustProjectReference(project string) string {
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_google_project_service.go:108]: This can be replaced with `GetResourceNameFromSelfLink`

**Review by rileykarson — APPROVED** [review]
Just a couple minor qusetions/nitpicks about self link related funcs!

File: third_party/terraform/resources/resource_google_project_service.go (hunk 1 above)
Comment (ScottSuarez) [reply third_party/terraform/resources/resource_google_project_service.go:89]: compareResourceNames is simpler under the hood. compareSelfLinkOrResourceName has a lot of complexity that I feel I didn't need. I tried to use it originally but ran into some issue. I try to switch to compareSelfLinkOrResourceName if you think that's the better practice

File: third_party/terraform/resources/resource_google_project_service.go (hunk 2 above)
Comment (ScottSuarez) [reply third_party/terraform/resources/resource_google_project_service.go:108]: Will do

File: third_party/terraform/resources/resource_google_project_service.go (hunk 1 above)
Comment (rileykarson) [reply third_party/terraform/resources/resource_google_project_service.go:89]: All good! just stood out that this was a non-standard one.

**Pull request merged**